src/main/java/org/toradocu/extractor/JavadocExtractor.java

```diff
@@ -1,515 +1,664 @@
-      List<String> stringCodeTags =
-          taggedComment.select("code").stream().map(Element::text).collect(Collectors.toList());
+    for (int i = 0; i < reflectionParams.length; i++) {
+      final java.lang.reflect.Parameter reflectionParam = reflectionParams[i];
+      final String reflectionQualifiedTypeName =
+          rawType(reflectionParam.getParameterizedType().getTypeName());
+      String reflectionSimpleTypeName = removePackage(reflectionQualifiedTypeName, false);
+
+      final com.github.javaparser.ast.body.Parameter sourceParam = sourceParams.get(i);
+      String sourceTypeName = rawType(sourceParam.getType().asString());
+      if (sourceParam.isVarArgs()) sourceTypeName += "[]";
+      if (reflectionParam.isVarArgs() && !reflectionSimpleTypeName.contains("[]"))
+        reflectionSimpleTypeName += "[]";
+      int dollar = reflectionSimpleTypeName.indexOf("$");
+      if (dollar != -1) {
+        if (sourceTypeName.contains(".")) {
+          // In case of Enum
+          reflectionSimpleTypeName = reflectionSimpleTypeName.replace("$", ".");
+        } else {
+          // In case of nested class
+          reflectionSimpleTypeName =
+              reflectionSimpleTypeName.substring(dollar + 1, reflectionSimpleTypeName.length());
+        }
+      }
 
-      ThrowsTag tagToProcess =
-          new ThrowsTag(
-              new org.toradocu.extractor.Type(getExceptionName(throwsTag, member)),
-              comment,
-              stringCodeTags);
-      memberThrowsTags.add(tagToProcess);
+      if (!reflectionSimpleTypeName.equals(sourceTypeName)) {
+        return false;
+      }
     }
 
-    return memberThrowsTags;
+    return true;
   }
 
   /**
-   * Returns all the @throws and @exception tags in the documentation of the given class member.
+   * Get raw type corresponding to a possibly-generic type; that is, remove generic type arguments.
    *
-   * @param member a class member
-   * @return a list with all the @throws and @exception tags of member
+   * @param type string representation of the type
+   * @return the raw type
    */
-  private static List<Tag> throwsTagsOf(Doc member) {
-    final List<Tag> throwsTags = new ArrayList<>();
-    Collections.addAll(throwsTags, member.tags("@exception"));
-    Collections.addAll(throwsTags, member.tags("@throws"));
-    return throwsTags;
+  private String rawType(String type) {
+    int i = type.indexOf("<");
+    if (i != -1) { // If type contains "<".
+      return type.substring(0, i);
+    }
+    return type;
   }
 
   /**
-   * @param member the constructor or method from which to extract the throws tags
-   * @param classDoc the class that we use to extract the tags
-   * @return the list that contains the ParamTags of the class we gave
-   * @throws IOException if the method encounters an error while reading/generating class
-   *     documentation
+   * Remove package from a type name. For example, given "java.lang.String", this method returns
```

Comment:
Document how this handles inner classes.  Also, what is the argument input format?